frontend/src/metabase/visualizations/components/ChartSettings.jsx

```diff
@@ -251,6 +269,7 @@ class ChartSettings extends Component {
       onShowWidget: this.handleShowWidget,
       onEndShowWidget: this.handleEndShowWidget,
       currentSectionHasColumnSettings,
+      columnHasSettings: col => this.columnHasSettings(col),
```

Comment:
Instead of passing a callback to compute this value, it should be possible to use `widget.value` and  `widget.props.columns` from `visibleWidgets` to compute this boolean flag here directly.

Reply:
Interesting. If it ends up being false, I could just set `showColumnSetting` to false. I'll give it a go